rasa/core/brokers/kafka.py

```diff
@@ -107,12 +105,14 @@ def publish(
         retry_delay_in_seconds: float = 5,
     ) -> None:
         """Publishes events."""
+        from confluent_kafka import KafkaException
+
         if self.producer is None:
             self.producer = self._create_producer()
-            connected = self.producer.bootstrap_connected()
-            if connected:
+            try:
+                self.producer.flush(timeout=1)
```

Comment:
did you check that it properly fails if there is nothing to flush? (just double checking that this always raises an exception if we can't connect and that there isn't a case where the call to `flush` is a no-op resulting in no exception. in analytics we used a different call to verify the connection so just want to make sure)

Reply:
i mean it might work 🤷 the method used in analytics is from stackoverflow 😉

Reply:
I think I chose the call to `flush` because the docs say that the error callback function gets called by `flush` in case of issues. Another alternative would be to use `list_topics` as in this [example](https://stackoverflow.com/a/61227038/15463002) 🤔